module/modmanager/manager.go

```diff
@@ -365,19 +365,14 @@ func (mgr *Manager) Reconfigure(ctx context.Context, conf config.Module) ([]reso
 	mgr.logger.CInfow(ctx, "New module process is running and responding to gRPC requests", "module",
 		mod.cfg.Name, "module address", mod.addr)
 
-	// add old module process' resources to new module; warn if new module cannot
-	// handle old resource and consider that resource orphaned.
```

Comment:
For clarity, this logic is now handled in `ResolveImplicitDependenciesInConfig` by the appension of resources that are managed by a module that is in `conf.Modified.Modules` to `conf.Added.Components/Services`. All of this code is... fragile (_perhaps_ necessarily), but I think it's better to make the appropriate changes to the `config.Diff` and let the resource manager handle the re-addition of modular resources to the new module process rather than add them here and force extra `ReconfigureResourceRequest`s later on once we've calculated implicit dependencies. If you're wondering, we cannot really resolve implicit dependencies at this point in the code; we _could_ run `ValidateConfig` to find the _names_ of the implicit dependencies, but the resource manager will not be aware of the edges in the resource graph that need to be added to account for said dependencies.